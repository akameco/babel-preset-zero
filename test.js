// @flow

const cases = require('jest-in-case')
const { transform } = require('@babel/core')

jest.mock('ptils', () => ({
  hasAnyDep: jest.fn(() => false),
}))

function snapshot(name, input) {
  const { code } = transform(input, {
    presets: [require.resolve('.')],
  })
  expect(code).toMatchSnapshot(name)
}

cases(
  'plugin snapshot',
  ({ name, input }) => {
    snapshot(name, input)
  },
  [
    { name: 'object reset spread', input: 'const a = {...foo}' },
    {
      name: 'class-properties',
      input: `
        class A { a = () => 'hello' }
      `,
    },
    {
      name: 'dead code',
      input: `
        function hoge() {var x = 1;}
      `,
    },
    {
      name: 'import',
      input: `
        import React from 'react';
      `,
    },
  ]
)

// cases(
// 'presets snapshot',
// ({ name, input }) => {
// snapshot(name, input)
// },
// [
// {
// name: 'flowtype',
// input: `
// // @flow
// type A = string
// const a: A = 'hello'`,
// },
// {
// name: 'react',
// input: `
// const A = () => <div>hello</div>
// `,
// },
// ]
// )
