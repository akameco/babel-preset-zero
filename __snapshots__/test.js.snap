// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`plugin snapshot class-properties: class-properties 1`] = `
"\\"use strict\\";

class A {
  constructor() {
    this.a = () => 'hello';
  }

}"
`;

exports[`plugin snapshot dead code: dead code 1`] = `
"\\"use strict\\";

function hoge() {}"
`;

exports[`plugin snapshot import: import 1`] = `
"\\"use strict\\";

var _react = _interopRequireDefault(require(\\"react\\"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }"
`;

exports[`plugin snapshot object reset spread: object reset spread 1`] = `
"\\"use strict\\";

const a = Object.assign({}, foo);"
`;
